packages/astro/src/core/redirects/helpers.ts

```diff
@@ -20,6 +20,16 @@ export function redirectRouteGenerate(redirectRoute: RouteData, data: Params): s
 	if (typeof routeData !== 'undefined') {
 		return routeData?.generate(data) || routeData?.pathname || '/';
 	} else if (typeof route === 'string') {
+		// TODO: this logic is duplicated between here and manifest/create.ts
+		if (Object.keys(data).length > 0) {
+			let target = route
+			for (const param in data) {
+				const paramValue = data[param]!
```

Comment:
nit: I don't like very much the use of `for..in`, we should use `for..of`

Reply:
its not an array

Reply:
```js
const dataAsArray  =Object.entries(data);

if (dataAsArray.length >) {
	for..of
}
```

Reply:
oh didnt see this, implemented using object.keys